core/arch/arm/kernel/generic_boot.c

```diff
@@ -662,3 +663,20 @@ void generic_boot_init_secondary(unsigned long nsec_entry)
 	init_secondary_helper(nsec_entry);
 }
 #endif
+
+#ifdef CFG_BOOT_SECONDARY_REQUEST
+int generic_smp_boot(size_t core_idx, uint32_t entry)
+{
+	if ((core_idx == 0) || (core_idx >= CFG_TEE_CORE_NB_CORE))
+		return -1;
+
+	/* set secondary cores' NS entry addresses */
+
+	ns_entry_addrs[core_idx] = entry;
+	spin_table[core_idx] = 1;
```

Comment:
i think you need a `dmb` so that spin is updated before ns entry address reach a cache.